src/Common/ShellCommand.cpp

```diff
@@ -335,5 +354,50 @@ void ShellCommand::wait()
     }
 }
 
+ShellCommand::Environ::Environ(const std::vector<std::string> *env_vars)
+{
+    data = nullptr;
+    if (!env_vars->empty())
+    {
+        // loosely lifted from the freebsd source
+        char ** envp;
+        for (envp = environ; envp && *envp; envp++) {}
+        size_t env_num = envp - environ;
+        orig_number_of_env_vars = env_num;
+        size_t needed_size = (env_num + env_vars->size() / 2 + 1) * sizeof(*envp);
+        data = static_cast<char **>(malloc(needed_size));
+        // here we only copy the pointers, as clickhouse should __never__ call setenv() when threads are up
```

Comment:
`__never__` sounds scary.

Reply:
`f()` is K&R style, a function with name `f` should be written as `f` in comments.